scheduler/services/resource/base.py

```diff
@@ -102,6 +102,17 @@ def __init__(self, sites: FrozenSet[Site] = ALL_SITES):
         # The final output from this class: the configuration per night.
         self._night_configurations: Dict[Site, Dict[date, NightConfiguration]] = {site: {} for site in self._sites}
 
+    @staticmethod
+    def _mdf_to_barcode(mdfname: str, inst: str) -> Resource:
+        """Legacy MOS mask barcode convention"""
+        barcode = None
+        instd = {'GMOS': '1', 'GMOS-N': '1', 'GMOS-S': '1', 'Flamingos2': '3'}
+        semd = {'A': '0', 'B': '1'}
+        progd = {'Q': '0', 'C': '1', 'L': '2', 'F': '3', 'S': '8', 'D': '9'}
```

Comment:
Since these are constants, I'm going to move them out of the method and into the class, but I can take care of that after.